### PR TITLE
Texnixe/feature layouts methods

### PR DESCRIPTION
**Pull request by texnixe**
## Describe the PR
Adds new `toBlocks()` and `hasBlockType()` Layouts methods. Useful if you need to check if the layouts field contains a given block type, to load scripts conditionally (similar to  new `hasBlock()` Blocks method). The `toBlocks()` method makes it easy to filter blocks from layouts.

### Features

- New `$layouts->toBlocks()` method
- new `$layouts->hasBlockType()` method

## Breaking changes
none

## Related issues/ideas
https://github.com/getkirby/kirby/pull/3555#pullrequestreview-712122102

## Ready?

- [x] Unit tests for fixed bug/feature
- [x] In-code documentation (wherever needed)
- [x] CI checks pass

## When merging

- [ ] Add to [website docs release checklist](https://github.com/getkirby/getkirby.com/pulls) (if needed)
- [x] Add changes to release notes draft in Notion

### CODE REVIEW

File: src/Cms/Layouts.php [hunk 1]
```diff
@@ -41,6 +41,19 @@ public static function factory(array $items = null, array $params = [])
...
+     * Checks if a given block type exists in the layouts collection
+     *
+     * @param string $type
+     * @return bool
+     */
+    public function hasBlockType(string $type): bool
+    {
+        $blocks = $this->toBlocks();
+
+        return $blocks->hasType($type);
```
Comment (afbora) [inline src/Cms/Layouts.php]:
```suggestion
        return $this->toBlocks()->hasType($type);
```

File: src/Cms/Layouts.php [hunk 2]
```diff
@@ -63,4 +76,25 @@ public static function parse($input): array
...
+        }
+        foreach ($this as $layout) {
+            foreach ($layout->columns() as $column) {
+                foreach ($column->blocks() as $block) {
+                    $blocks[] = $block->toArray();
+                }
+            }
+        }
+
+        return Blocks::factory($blocks);
```
Comment (afbora) [inline src/Cms/Layouts.php]:
```suggestion
    /**
     * Converts layouts to blocks
     *
     * @return \Kirby\Cms\Blocks
     */
    public function toBlocks()
    {
        $blocks = [];

        if ($this->isNotEmpty() === true) {
            foreach ($this->data() as $layout) {
                foreach ($layout->columns() as $column) {
                    foreach ($column->blocks() as $block) {
                        $blocks[] = $block->toArray();
                    }
                }
            }
        }

        return Blocks::factory($blocks);
    }
```

File: src/Cms/Layouts.php [hunk 3]
```diff
@@ -78,21 +78,23 @@ public static function parse($input): array
...
+                    foreach ($column->blocks() as $block) {
+                        $blocks[] = $block->toArray();
+                    }
                 }
             }
         }
 
         return Blocks::factory($blocks);
     }
+    }
```
Comment (afbora) [inline src/Cms/Layouts.php]: Oh my mistake on suggestion lines, sorry. You can fix that with removing this line and squash them.

Comment (afbora) [reply src/Cms/Layouts.php]: Thank you @texnixe for fixing that. Could you squash all commits to single?

**Review by bastianallgeier — DISMISSED** [review]

**Review by afbora — APPROVED** [review]

**Pull request merged**